TestFramework/ExtensionMethods.cs

```diff
@@ -381,5 +377,59 @@ public static bool WaitForPageLoad(IWebDriver driver, By by, ScenarioContext sce
                 return false;
             }
         }
+
+        public static void WaitForElementNotVisible(IWebDriver driver, By by, int? timeInSec = null)
+        {
+            bool isVisible = true;
+            timeInSec = timeInSec == null ? 20 : timeInSec.Value;
+            int count = 1;
+            while (isVisible)
+            {
+                try
+                {
+                    if (!IsElementVisible(driver, by, null))
+                    {
+                        break;
+                    }
+                    WebDriverWait wait = new WebDriverWait(driver, TimeSpan.FromSeconds(1));
+                    wait.Until(ExpectedConditions.InvisibilityOfElementLocated(by));
+                    isVisible = IsElementVisible(driver, by, null);
+                }
+                catch
+                {
+
+                }
+                if (count > timeInSec && isVisible)
+                {
+                    throw new Exception($"Element {by.Criteria} was not expected to be visible after {timeInSec} sec, but it was");
```

Comment:
I see the exception is thrown and not handled anywhere. It's recommended we also log this